crates/nu-engine/src/documentation.rs

```diff
@@ -27,70 +32,216 @@ pub fn get_full_help(
     // execution.
     let stack = &mut stack.start_collect_value();
 
-    let signature = engine_state
+    let nu_config = stack.get_config(engine_state);
+
+    let sig = engine_state
         .get_signature(command)
         .update_from_command(command);
 
-    get_documentation(
-        &signature,
-        &command.examples(),
+    // Create ansi colors
+    let mut help_style = HelpStyle::default();
+    help_style.update_from_config(engine_state, &nu_config);
+
+    let mut long_desc = String::new();
+
+    let desc = &sig.description;
+    if !desc.is_empty() {
+        long_desc.push_str(&highlight_code(desc, engine_state, stack));
+        long_desc.push_str("\n\n");
+    }
+
+    let extra_desc = &sig.extra_description;
+    if !extra_desc.is_empty() {
+        long_desc.push_str(&highlight_code(extra_desc, engine_state, stack));
+        long_desc.push_str("\n\n");
+    }
+
+    match command.command_type() {
+        CommandType::Alias => get_alias_documentation(
+            &mut long_desc,
+            command,
+            &sig,
+            &help_style,
+            engine_state,
+            stack,
+        ),
+        _ => get_command_documentation(
+            &mut long_desc,
+            command,
+            &sig,
+            &nu_config,
+            &help_style,
+            engine_state,
+            stack,
+        ),
+    };
+
+    if !nu_config.use_ansi_coloring.get(engine_state) {
+        nu_utils::strip_ansi_string_likely(long_desc)
+    } else {
+        long_desc
+    }
+}
+
+/// Syntax highlight code using the `nu-highlight` command if available
+fn try_nu_highlight(
+    code_string: &str,
+    reject_garbage: bool,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+) -> Option<String> {
+    let highlighter = engine_state.find_decl(b"nu-highlight", &[])?;
+
+    let decl = engine_state.get_decl(highlighter);
+    let mut call = Call::new(Span::unknown());
+    if reject_garbage {
+        call.add_named((
+            Spanned {
+                item: "reject-garbage".into(),
+                span: Span::unknown(),
+            },
+            None,
+            None,
+        ));
+    }
+
+    decl.run(
         engine_state,
         stack,
-        command.is_keyword(),
+        &(&call).into(),
+        Value::string(code_string, Span::unknown()).into_pipeline_data(),
     )
+    .and_then(|pipe| pipe.into_value(Span::unknown()))
+    .and_then(|val| val.coerce_into_string())
+    .ok()
 }
```

Comment:
I see what you mean. That is really convoluted and shouldn't happen in our code. But yeah, isolating `nu-parser` would be the way to go and that gets tough.

Reply:
it seems like it might actually be realistic to remove nu-engine as a dependency from the parser (which I assume we will need for `new-nu-parser` anyway), i can investigate that if i get the chance but we can move forward with this as-is

Reply:
I think that's out of scope of this PR